plugins/catalog-backend-module-github/src/providers/GitHubOrgEntityProvider.ts

```diff
@@ -88,6 +88,16 @@ export interface GitHubOrgEntityProviderOptions {
    * Optionally supply a custom credentials provider, replacing the default one.
    */
   githubCredentialsProvider?: GithubCredentialsProvider;
+
+  /**
+   * Optionally use the namespace to use in the catalog for users from this org.
+   */
+  userNamespace?: string;
```

Comment:
This is a bit of an unrelated feature to this PR right?

A pattern that we have used in other places, e.g. [in the LDAP org ingestion provider](https://github.com/backstage/backstage/blob/7506c95e94e94ba8b34e6ef4144cafc59fc15258/plugins/catalog-backend-module-ldap/src/processors/LdapOrgEntityProvider.ts#L86), is to be able to supply a optional transformers for users and groups.

Maybe something similar could be added to this provider as well. The callback would be provided with all of the available data out of the requests to github, and perhaps it could export a [default transformer](https://github.com/backstage/backstage/blob/7506c95e94e94ba8b34e6ef4144cafc59fc15258/plugins/catalog-backend-module-ldap/src/ldap/read.ts#L44) that is easy to call. Then you could do something like

```ts
new GitHubOrgEntityProvider({
  // ...
  async userTransformer(userdata): Promise<UserEntity | undefined> {
    const user = defaultUserTransformer(userdata);
    user.metadata.namespace = 'my-custom-namespace';
    return user;
  }
});
```

The reason I'm suggesting this is, there's a myriad of different variations that one could possibly want to do on the resulting entity, so adding one-off fields like this has less value over time.

And related to the above comment - among the userdata would then be both the org emails and the other email, so you can pick yourself which one to use, and possibly even based on some other condition.

But, in any case - would probably prefer to add this feature in a separate PR, not with the org email one.

Reply:
The reason I made this change was to enable the call to the `getOrganizationUsers` function to be explicit. This is because the `userNamespace` property is already optional without a default so I must pass something into it if I want to add another parameter. So, I made it configurable rather than just only adding default namespace internally to the function

Currently it just seems to be an implementation detail that its set to default somewhere rather than being explicit.

There are other approaches I could have added like an overload of the function that sets the default namespace. Or I can change the signature of the function entirely and fix all the places that call it. But then they all would probably have to pass in the default namespace explicitly rather than it happening explicitly somewhere else.